core/src/main/java/org/jboss/pnc/build/finder/core/BuildFinderUtils.java

```diff
@@ -374,4 +375,38 @@ private static List<String> getArchiveExtensionsFromKoji(BuildConfig config, Cli
 
         return Collections.unmodifiableList(extensionsToCheck);
     }
+
+    public Map<Checksum, Collection<String>> swapEntriesWithPreferredChecksum(
+            Map<Checksum, Collection<String>> originalMap,
+            Map<String, Collection<Checksum>> fileInverseMap,
+            ChecksumType preferredChecksumType) {
+
+        Map<Checksum, Collection<String>> preferredChecksumMap = new HashMap<>(originalMap.size(), 1.0f);
```

Comment:
Because we know in advance how many elements the map will contain, and we want to minimize resizing...but I guess it's a tiny performance improvement.